ceph/rados/crushtool_workflows.py

```diff
@@ -0,0 +1,649 @@
+"""
+Module to Generate, Modify and Apply Crush maps on ceph cluster. Methods include:
+1. Generate bin file.
+2. Compile bin file
+3. decomiple bin file
+4. setting the new bin file on cluster.
+5. Addition of new bucket into bin file, and it's verification.
+6. Re-weighting the OSDs in bin file, and it's verification.
+7. Moving buckets from one to other in bin file, and it's Verification.
+8. bin file tests. ( stats, bad mappings etc.)
+9. dump and verify bin file contents
+"""
+import json
+import re
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def count(func):
+    """
+    Decorator method to check how many times a particular method has been invoked
+    :param func: name of the function
+    :return: wrapped method
+    """
+
+    def wrapped(*args, **kwargs):
+        wrapped.calls += 1
+        return func(*args, **kwargs)
+
+    wrapped.calls = 0
+    return wrapped
+
+
+class CrushToolWorkflows:
+    """
+    Contains various functions that help to Generate, Modify and Apply Crush maps on ceph cluster
+    """
+
+    def __init__(self, node: CephAdmin):
+        """
+        initializes the env to run CRUSH commands
+
+        Initializes the crush object with the cluster objects,along with installing necessary packages
+        required for executing crushtool commands
+        Args:
+            node: CephAdmin object
+        """
+        self.rados_obj = RadosOrchestrator(node=node)
+        self.cluster = node.cluster
+        self.client = node.cluster.get_nodes(role="client")[0]
+
+        # Checking and installing ceph-base package on Client
+        out, rc = self.client.exec_command(sudo=True, cmd="rpm -qa | grep ceph-base")
```

Comment:
Excatly, this is just the check to see if the package exists, else it will be installed. This module is supposed to be used independently from the suite file. Hence the check, and if fail, it installs the package.